test/Faker/Provider/ImageTest.php

```diff
@@ -45,18 +45,30 @@ public function testUrlWithDimensionsAndBadCategory()
 
     public function testDownloadWithDefaults()
     {
-        $file = Image::image(sys_get_temp_dir());
-        $this->assertFileExists($file);
-        if (function_exists('getimagesize')) {
-            list($width, $height, $type, $attr) = getimagesize($file);
-            $this->assertEquals(640, $width);
-            $this->assertEquals(480, $height);
-            $this->assertEquals(constant('IMAGETYPE_JPEG'), $type);
-        } else {
-            $this->assertEquals('jpg', pathinfo($file, PATHINFO_EXTENSION));
-        }
-        if (file_exists($file)) {
-            unlink($file);
+        $url = "http://www.lorempixel.com/";
+        $curlPing = curl_init($url);
+        curl_setopt($curlPing, CURLOPT_TIMEOUT, 5);
+        curl_setopt($curlPing, CURLOPT_CONNECTTIMEOUT, 5);
+        curl_setopt($curlPing, CURLOPT_RETURNTRANSFER, true);
+        $data = curl_exec($curlPing);
+        $httpCode = curl_getinfo($curlPing, CURLINFO_HTTP_CODE);
+        curl_close($curlPing);
+
+        if ($httpCode >= 200 && $httpCode < 300)
```

Comment:
I suggest you reverse the logic:

```php
if ($httpCode < 200 || $httpCode > 300) {
   $this-> markTestSkipped('LoremPixel is offline, skipping image download');
  return;
}
```

cf. http://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html